compiler/runtime.go

```diff
@@ -541,7 +546,7 @@ func (e *callEvaluator) Eval(scope Scope) (values.Value, error) {
 	if err != nil {
 		return nil, err
 	}
-	return f.Function().Call(args.Object())
+	return f.Function().Call(e.ctx, e.deps, args.Object())
```

Comment:
Should the context and deps come from `Eval` here instead of being compiled in?

In other words, should we revert `complier/Compile` to not know about context or dep but instead teach the `runtime/Eval` interface about them? This way we do not have to worry about caching bad context or deps as they are passed in with each run of the runtime.